sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SerializerBuildHelper.scala

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+import org.apache.spark.sql.catalyst.expressions.{CreateNamedStruct, Expression, IsNull, UnsafeArrayData}
+import org.apache.spark.sql.catalyst.expressions.objects._
+import org.apache.spark.sql.catalyst.util.{DateTimeUtils, GenericArrayData}
+import org.apache.spark.sql.types._
+import org.apache.spark.unsafe.types.UTF8String
+
+object SerializerBuildHelper {
+
+  def createSerializerForBoolean(inputObject: Expression): Expression = {
+    Invoke(inputObject, "booleanValue", BooleanType)
+  }
+
+  def createSerializerForByte(inputObject: Expression): Expression = {
+    Invoke(inputObject, "byteValue", ByteType)
+  }
+
+  def createSerializerForShort(inputObject: Expression): Expression = {
+    Invoke(inputObject, "shortValue", ShortType)
+  }
+
+  def createSerializerForInteger(inputObject: Expression): Expression = {
+    Invoke(inputObject, "intValue", IntegerType)
+  }
+
+  def createSerializerForLong(inputObject: Expression): Expression = {
+    Invoke(inputObject, "longValue", LongType)
+  }
+
+  def createSerializerForFloat(inputObject: Expression): Expression = {
+    Invoke(inputObject, "floatValue", FloatType)
+  }
+
+  def createSerializerForDouble(inputObject: Expression): Expression = {
+    Invoke(inputObject, "doubleValue", DoubleType)
+  }
```

Comment:
Seems these helper methods don't reduce code and just add one more wrapper around calling `Invoke`. Are they needed?

Reply:
This is not for reducing code. This is for consistency. These methods ensure we are consistently serialize / deserialize things between ScalaReflection and JavaTypeInference if the type is same.